File: SpeedGui/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Microsoft.FSharp;
using Microsoft.FSharp.Core;

using Speed;
using Spc = Speed.Core;
using Spct = Speed.Core.Types;

namespace SpeedGui
{
    public partial class MainForm : Form
    {
        public MainForm()
        {
            InitializeComponent();
        }

        Spct.GameT<Spct.PlayerT<Unit, Unit>> gs_;

        private void DrawTitle(Graphics gx)
        {
            gx.DrawString
                ("Speed F#"
                , new Font("Yu Gothic", 35)
                , new SolidBrush(Color.Black)
                , new PointF(30, 50)
                );

            gx.DrawString
                ("Click to Start!"
                , new Font("Yu Gothic", 16)
                , new SolidBrush(Color.Black)
                , new PointF(70, 150)
                );
        }

        private void DrawBoard(Graphics gx)
        {
        }

        private void StartGame()
        {
            // unimplemented
        }

        private void MainForm_Paint(object sender, PaintEventArgs e)
        {
            DrawTitle(e.Graphics);
        }

        private void MainForm_Click(object sender, EventArgs e)
        {
            StartGame();
        }
    }
}

File: SpeedGui/MainForm.fs
namespace Speed.Gui

open System
open System.Drawing
open System.Windows.Forms
open Speed
open Speed.Core

[<AutoOpen>]
module Misc =
  let yuGothic emSizeInt =
    new Font("Yu Gothic", emSizeInt |> float32)
  let pointF x y =
    new PointF(float32 x, float32 y)
  let blackBrush =
    new SolidBrush(System.Drawing.Color.Black)

  type MainFormState =
    | BeforeGame
    | InGame      of GameState
    | AfterGame   of GameResult

type MainForm () =
  inherit Form
    ( Text = "Speed F#"
    )

  /// Some g iff game is running
  let mutable cur = BeforeGame

  let drawTitleScreen (gfx: Graphics) =
    do
      gfx.DrawString
        ("Speed F#"
        , yuGothic 35
        , blackBrush
        , pointF 30 50
        )
    do
      gfx.DrawString
        ("Click to Start!"
        , yuGothic 16
        , blackBrush
        , pointF 70 150
        )

  override this.OnPaint(e) =
    match cur with
    | BeforeGame ->
        drawTitleScreen(e.Graphics)
    | _ ->
        ()
